File: src/nl/rubensten/texifyidea/gutter/LatexLineMarkerProvider.java
package nl.rubensten.texifyidea.gutter;

import com.intellij.codeInsight.daemon.RelatedItemLineMarkerInfo;
import com.intellij.codeInsight.daemon.RelatedItemLineMarkerProvider;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;

import java.util.Collection;

/**
 * @author Sten Wessel
 */
public class LatexLineMarkerProvider extends RelatedItemLineMarkerProvider {

    @Override
    protected void collectNavigationMarkers(@NotNull PsiElement element,
                                            Collection<? super RelatedItemLineMarkerInfo> result) {
        new LatexNavigationGutter().collectNavigationMarkers(element, result);
        new LatexCompileGutter().collectNavigationMarkers(element, result);
    }
}

File: src/nl/rubensten/texifyidea/gutter/LatexLineMarkerProvider.kt
package nl.rubensten.texifyidea.gutter

import com.intellij.codeInsight.daemon.RelatedItemLineMarkerInfo
import com.intellij.codeInsight.daemon.RelatedItemLineMarkerProvider
import com.intellij.psi.PsiElement

/**
 * @author Sten Wessel
 */
open class LatexLineMarkerProvider : RelatedItemLineMarkerProvider() {

    override fun collectNavigationMarkers(element: PsiElement,
                                          result: MutableCollection<in RelatedItemLineMarkerInfo<PsiElement>>?) {
        LatexNavigationGutter().collectNavigationMarkers(element, result)
    }
}
